From 7066ff041cec66e222845e549d91257708a635bf Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 18 May 2026 07:45:35 +0000
Subject: [PATCH] perf/x86/intel/uncore: Fix reference count leak in
 hswep_has_limit_sbox()

jira VULN-154139
cve CVE-2022-50318
commit-author Xiongfeng Wang <wangxiongfeng2@huawei.com>
commit 1ff9dd6e7071a561f803135c1d684b13c7a7d01d

pci_get_device() will increase the reference count for the returned
'dev'. We need to call pci_dev_put() to decrease the reference count.
Since 'dev' is only used in pci_read_config_dword(), let's add
pci_dev_put() right after it.

Fixes: 9d480158ee86 ("perf/x86/intel/uncore: Remove uncore extra PCI dev HSWEP_PCI_PCU_3")
	Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
	Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
	Reviewed-by: Kan Liang <kan.liang@linux.intel.com>
Link: https://lore.kernel.org/r/20221118063137.121512-3-wangxiongfeng2@huawei.com
(cherry picked from commit 1ff9dd6e7071a561f803135c1d684b13c7a7d01d)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 arch/x86/events/intel/uncore_snbep.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/events/intel/uncore_snbep.c b/arch/x86/events/intel/uncore_snbep.c
index e9195fec8b38..9ea24b0f121b 100644
--- a/arch/x86/events/intel/uncore_snbep.c
+++ b/arch/x86/events/intel/uncore_snbep.c
@@ -2891,6 +2891,7 @@ static bool hswep_has_limit_sbox(unsigned int device)
 		return false;
 
 	pci_read_config_dword(dev, HSWEP_PCU_CAPID4_OFFET, &capid4);
+	pci_dev_put(dev);
 	if (!hswep_get_chop(capid4))
 		return true;
 
-- 
2.47.3

